controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller_test.go

```diff
@@ -0,0 +1,762 @@
+		Expect(hostingdeploymentautoscalingjob.Status.HostingDeploymentAutoscalingJobStatus).To(Equal(CreatedAutoscalingJobStatus))
+		Expect(hostingdeploymentautoscalingjob.Status.PolicyName).To(Equal(policyName))
+		// TODO mbaijal: Add PolicyARN ?
+		//Expect(hostingdeploymentautoscalingjob.Status.policyArn).To(Equal(policyArn))
+	})
+
+})
+
+var _ = Describe("Reconciling an HAP that is different from the spec", func() {
+
+	var (
+		receivedRequests                                     List
+		mockAutoscalingClientBuilder                         *MockAutoscalingClientBuilder
+		hostingdeploymentautoscalingjob                      *hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJob
+		outOfDateTargetDescription                           applicationautoscaling.DescribeScalableTargetsOutput
+		outOfDatePolicyDescription, updatedPolicyDescription applicationautoscaling.DescribeScalingPoliciesOutput
+		controller                                           Reconciler
+		request                                              ctrl.Request
+		policyName, policyArn                                string
+	)
+
+	BeforeEach(func() {
+		hostingdeploymentautoscalingjob = createHAPWithFinalizer()
+		err := k8sClient.Create(context.Background(), hostingdeploymentautoscalingjob)
+		Expect(err).ToNot(HaveOccurred())
+
+		policyName = "test-policy-name"
+		policyArn = "policy-arn"
+		resourceID := "endpoint/endpoint-xyz/variant/variant-xyz"
+		scalableTarget := applicationautoscaling.ScalableTarget{
+			ResourceId: &resourceID,
+		}
+		scalableTargets := []applicationautoscaling.ScalableTarget{scalableTarget}
+		scalingPolicy := applicationautoscaling.ScalingPolicy{
+			PolicyName: &policyName,
+			PolicyARN:  &policyArn,
+			ResourceId: &resourceID,
+		}
+		scalingPolicies := []applicationautoscaling.ScalingPolicy{scalingPolicy}
+
+		outOfDateTargetDescription = applicationautoscaling.DescribeScalableTargetsOutput{
+			ScalableTargets: scalableTargets,
+		}
+		outOfDatePolicyDescription = applicationautoscaling.DescribeScalingPoliciesOutput{
+			ScalingPolicies: scalingPolicies,
+		}
+		updatedPolicyDescription = applicationautoscaling.DescribeScalingPoliciesOutput{
+			ScalingPolicies: scalingPolicies,
+		}
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Get(context.Background(), types.NamespacedName{
+			Name:      hostingdeploymentautoscalingjob.ObjectMeta.Name,
+			Namespace: hostingdeploymentautoscalingjob.ObjectMeta.Namespace,
+		}, hostingdeploymentautoscalingjob)
+		Expect(err).ToNot(HaveOccurred())
+
+		hostingdeploymentautoscalingjob.ObjectMeta.Finalizers = []string{}
+		err = k8sClient.Update(context.Background(), hostingdeploymentautoscalingjob)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Mark job as deleting in Kubernetes.
+		err = k8sClient.Delete(context.Background(), hostingdeploymentautoscalingjob)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("When the delete and creation succeeds", func() {
```

Comment:
I don't understand this context message

Reply:
I think the intend is a create can follow a delete only if the delete was successful ? It's the same as what model was testing but I agree all the unit test messages need a bit of rework. 
I'll try to fix a few for now.

Reply:
update includes a delete and then a re-create